helper/chart_data.js

```diff
@@ -0,0 +1,2932 @@
+                    question: element.event.questionName,
+                    responseType:element.event.questionResponseType,
+                    answers: [element.event.questionResponseLabel],
+                    chart: {},
+                    instanceQuestions: [],
+                    criteriaName: element.event.criteriaName,
+                    criteriaId: element.event.criteriaId,
+                    optionsAvailableForUser:element.event.answerOptions
+                }
+
+                // if(element.event.remarks != null){
+                //     resp.remarks = [element.event.remarks]
+                // }
+
+                result.response.push(resp);
+
+            }
+
+        }))
+
+        //filter all the objects whose questionResponseType is multiselect
+        await Promise.all(data.map(element => {
+            if (element.event.questionResponseType == "multiselect" && element.event.instanceParentResponsetype != "matrix") {
+                multiSelectArray.push(element);
+            }
+            if (element.event.instanceParentResponsetype == "matrix" && element.event.questionAnswer != null) {
+                matrixArray.push(element);
+            }
+        }))
+
+        //group the multiselect questions based on their questionExternalId
+        let multiSelectResult = await groupArrayByGivenField(multiSelectArray, "questionExternalId");
+        let res = Object.keys(multiSelectResult);
+
+        //loop the keys and construct a response object for multiselect questions
+        await Promise.all(res.map(async ele => {
+            let multiSelectResp = await instanceMultiselectFunc(multiSelectResult[ele])
+            result.response.push(multiSelectResp);
+
+        }))
+
+        console.log(matrixArray,'matrixArray')
+
+        //group the Matrix questions based on their questionExternalId
+        let matrixResult = await groupArrayByGivenField(matrixArray, "instanceParentId");
+        let matrixRes = Object.keys(matrixResult);
+
+        //loop the keys of matrix array
+        await Promise.all(matrixRes.map(async ele => {
+            let matrixResponse = await matrixResponseObjectCreateFunc(matrixResult[ele], solutionType)
+            result.response.push(matrixResponse);
+
+        }))
+
+        //sort the response objects based on questionExternalId field
+        await result.response.sort(getSortOrder("order")); //Pass the attribute to be sorted on
+
+        if (!reportType || reportType == filesHelper.survey) {
+            // Get the questions array
+            let questionArray = await questionListObjectCreation(actualData);
+            result.allQuestions = questionArray;
+        }
+
+        //return final response object
+        return result;
+    }
+    catch (err) {
+        console.log(err);
+    }
+}
+
+
+//Function to create a response object for multiselect question (Instance level Report)
+async function instanceMultiselectFunc(data) {
```

Comment:
@borkarsaish65 Please add proper function description ( Please check for all other functions in this helper)

Reply:
this is removed as it was not used yet and was moved frm ml-survey